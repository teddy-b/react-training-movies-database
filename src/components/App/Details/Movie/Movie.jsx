/* @flow */

import * as React from 'react'

import {
  StyledMovieSection,
  StyledMovie,
  StyledMoviePoster,
  StyledMovieDetails,
  StyledDetailsHeader,
  StyledHeading,
  StyledVoteAverage,
  StyledYear
} from './StyledMovie'
import Logo from '../../../shared/Logo'
import StyledImg from '../../../shared/styled/StyledImg'
import StyledLink from '../../../shared/styled/StyledLink'

type Props = {
  overview: string,
  poster_path: ?string,
  release_date: string,
  runtime: ?number,
  tagline: string,
  title: string,
  vote_average: number
}

const Movie = (props: Props): React.Node => {
  const {
    overview,
    poster_path: posterPath,
    release_date: releaseDate,
    runtime,
    tagline,
    title,
    vote_average: voteAverage
  } = props
  const year = new Date(releaseDate).getFullYear() || ''
  return (
    <StyledMovieSection>
      <StyledLink to="/">
        <Logo />
      </StyledLink>
      <StyledMovie>
        {posterPath &&
          <StyledMoviePoster>
            <StyledImg src={posterPath} alt={title} />
          </StyledMoviePoster>
        }
        <StyledMovieDetails>
          <StyledDetailsHeader>
            <StyledHeading>{title}</StyledHeading>
            <StyledVoteAverage>{voteAverage}</StyledVoteAverage>
          </StyledDetailsHeader>
          <h4>{tagline}</h4>
          <StyledYear>{year}</StyledYear>
          {runtime && <span>{runtime} min</span>}
          <p>{overview}</p>
        </StyledMovieDetails>
      </StyledMovie>
    </StyledMovieSection>
  )
}

export default Movie
