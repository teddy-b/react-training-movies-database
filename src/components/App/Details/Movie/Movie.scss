@import '../../../../styles/constants.scss';

.movieSection {
  background-color: $dimgrey;
  color: $white;
  padding: .5em;

  .movie {
    display: flex;
    justify-content: space-between;
  }
  
  .moviePoster {
    margin: 2%;
    width: 26%;
  
    img {
      width: 100%;
    }
  }
  
  .movieDetails {
    margin: 2%;
    width: 66%;

    .heading {
      display: flex;
    }

    .voteAverage {
      
    }
  
    h2 {
      color: $red;
    }
  
    span {
      padding-right: 1em;
    }
  }
}
