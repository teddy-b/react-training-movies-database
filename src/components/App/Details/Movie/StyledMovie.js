import styled from 'styled-components'

import { grey4, green, landscapePhoneMinWidth, red, white } from '../../../../constants/styles'

export const StyledMovieSection = styled.div`
  background-color: ${grey4};
  color: ${white};
  padding: .5em;
`

export const StyledMovie = styled.figure`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin: 0;

  @media screen and (min-width: ${landscapePhoneMinWidth}) {
    flex-wrap: nowrap;
  }
`

export const StyledMoviePoster = styled.div`
  margin: 2%;
  width: 98%;

  @media screen and (min-width: ${landscapePhoneMinWidth}) {
    width: 26%;
  }
`

export const StyledImg = styled.img`
  width: 100%;
`

export const StyledMovieDetails = styled.figcaption`
  margin: 2%;
  width: 98%;

  @media screen and (min-width: ${landscapePhoneMinWidth}) {
    width: 66%;
  }
`

export const StyledDetailsHeader = styled.div`
  display: flex;
  align-items: center;
`

export const StyledHeading = styled.h2`
  color: ${red};
`

export const StyledVoteAverage = styled.span`
  margin: 1em;
  padding: .5em;
  color: ${green};
  border: 2px solid ${green};
  border-radius: 50px;
`

export const StyledYear = styled.span`
  padding-right: 1em;
`
