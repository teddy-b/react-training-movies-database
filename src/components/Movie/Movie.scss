.movieSection {
  background-color: dimgrey;
  color: white;
  padding: .5em;
}

.movie {
  display: flex;
  justify-content: space-between;
}

.moviePoster {
  margin: 2%;
  width: 26%;

  img {
    width: 100%;
  }
}

.movieDetails {
  margin: 2%;
  width: 66%;

  h2 {
    color: red;
  }

  span {
    padding-right: 1em;
  }
}

.filmsBy {
  background-color: lightgrey;
  padding: .5em;
}
