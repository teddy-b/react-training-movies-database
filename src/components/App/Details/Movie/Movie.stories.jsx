import React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { storiesOf } from '@storybook/react'

import Movie from './Movie'
import moviesMock from '../../../../mocks/movies-mocks'

const singleMovieMock = moviesMock.data[0]

storiesOf('Components/Movie', module)
  .add('default', () => (
    <MemoryRouter>
      <Movie {...singleMovieMock} />
    </MemoryRouter>
  ))
