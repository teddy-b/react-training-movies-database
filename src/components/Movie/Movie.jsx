import React from 'react'

import Logo from '../Logo'
import './Movie.scss'

const Movie = ({ genres, overview, poster_path, release_date, runtime, tagline, title, vote_average }) => (
  <div>
    <div className="movieSection">
      <Logo />
      <figure className="movie">
        <div className="moviePoster">
          <img src={poster_path} alt={title} />
        </div>
        <figcaption className="movieDetails">
          <h2>{title}</h2>
          <h4>{tagline}</h4>
          <span>{new Date(release_date).getFullYear()}</span>
          <span>{runtime} min</span>
          <p>{overview}</p>
        </figcaption>
      </figure>
    </div>
    <div className="filmsBy">Films by {genres[0]} genre</div>
  </div>
  
)

export default Movie
