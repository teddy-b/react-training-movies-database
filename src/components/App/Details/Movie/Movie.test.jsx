/* @flow */

import * as React from 'react'
import { MemoryRouter } from 'react-router-dom'

import renderer from 'react-test-renderer'

import Movie from './Movie'
import moviesMock from '../../../../mocks/movies-mocks'

jest.mock('../../../shared/Logo', (): string => 'Logo')

describe('Movie', () => {
  const singleMovieMock = moviesMock.data[0]

  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Movie {...singleMovieMock} />
      </MemoryRouter>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with invalid date', () => {
    const releaseDate = ''
    const component = renderer.create(
      <MemoryRouter>
        <Movie
          {...singleMovieMock}
          release_date={releaseDate}
        />
      </MemoryRouter>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
