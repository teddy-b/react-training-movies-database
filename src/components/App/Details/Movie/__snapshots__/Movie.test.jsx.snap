// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `
<div
  className="sc-bdVaJa ilUIgN"
>
  <a
    className="sc-dnqmqq kxhpGu"
    href="/"
    onClick={[Function]}
  >
    <Logo />
  </a>
  <figure
    className="sc-bwzfXH haFnbb"
  >
    <div
      className="sc-htpNat iMSUrM"
    >
      <img
        alt="Fifty Shades Freed"
        className="sc-htoDjs kKCJWi"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </div>
    <figcaption
      className="sc-bxivhb dRhjdJ"
    >
      <div
        className="sc-ifAKCX dCoIjQ"
      >
        <h2
          className="sc-EHOje ixikkM"
        >
          Fifty Shades Freed
        </h2>
        <span
          className="sc-bZQynM jCBbRc"
        >
          6.1
        </span>
      </div>
      <h4>
        Don't miss the climax
      </h4>
      <span
        className="sc-gzVnrw igxgss"
      >
        2018
      </span>
      <span>
        106
         min
      </span>
      <p>
        Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
      </p>
    </figcaption>
  </figure>
</div>
`;

exports[`Movie renders correctly with invalid date 1`] = `
<div
  className="sc-bdVaJa ilUIgN"
>
  <a
    className="sc-dnqmqq kxhpGu"
    href="/"
    onClick={[Function]}
  >
    <Logo />
  </a>
  <figure
    className="sc-bwzfXH haFnbb"
  >
    <div
      className="sc-htpNat iMSUrM"
    >
      <img
        alt="Fifty Shades Freed"
        className="sc-htoDjs kKCJWi"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </div>
    <figcaption
      className="sc-bxivhb dRhjdJ"
    >
      <div
        className="sc-ifAKCX dCoIjQ"
      >
        <h2
          className="sc-EHOje ixikkM"
        >
          Fifty Shades Freed
        </h2>
        <span
          className="sc-bZQynM jCBbRc"
        >
          6.1
        </span>
      </div>
      <h4>
        Don't miss the climax
      </h4>
      <span
        className="sc-gzVnrw igxgss"
      >
        
      </span>
      <span>
        106
         min
      </span>
      <p>
        Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
      </p>
    </figcaption>
  </figure>
</div>
`;
